utils/python/tests/scripts_regression_tests.py

```diff
@@ -41,13 +41,41 @@ def test_resolve_variable_name(self):
         self.assertTrue(os.path.isfile(machinefile),
                         msg="Path did not resolve to existing file %s" % machinefile)
 
-    def test_CIME_unit_tests(self):
-        # Find and run all the unit tests in the CIME directory tree
+    def test_unittests(self):
+        # Finds all files contained in LIB_DIR or its subdirectories that match
+        # the pattern 'test*.py', and runs the unit tests found there (i.e.,
+        # tests defined using python's unittest module).
+        #
+        # This is analogous to running:
+        #     python -m unittest discover
+        #
+        # Yes, that means we have a bunch of unit tests run from this one unit
+        # test.
+
+        testsuite = unittest.defaultTestLoader.discover(
+            start_dir = LIB_DIR,
+            pattern = 'test*.py')
+
+        testrunner = unittest.TextTestRunner(buffer=False)
+
+        # Disable logging; otherwise log messages written by code under test
+        # clutter the unit test output
+        log_lvl = logging.getLogger().getEffectiveLevel()
+        logging.disable(logging.CRITICAL)
+        try:
+            results = testrunner.run(testsuite)
```

Comment:
Note that this produces output like this:

```
test_unittests (__main__.A_RunUnitTests) ... .............
----------------------------------------------------------------------
Ran 13 tests in 0.021s

OK
ok
```

According to earlier discussions with @jgfouca , it sounded like this bit of extra output is okay.